src/borg/testsuite/__init__.py

```diff
@@ -54,6 +55,86 @@
     st_mtime_ns_round = -4  # only >1 microsecond resolution here?
 
 
+_symlinks_supported = None
+_fifos_supported = None
+_hardlinks_supported = None
+_utime_supported = None
+
+
+def generate_tempfile_path():
+    return os.path.join(tempfile.gettempdir(), 'borg-test-' + uuid.uuid4().hex)
```

Comment:
I'm going to use `mktemp` because it was deprecated for security reasons which we don't need to worry about for this. Besides, this custom function has the same problems.